internal/testing/e2e/consul.go

```diff
@@ -83,21 +83,23 @@ func init() {
 }
 
 type consulTestEnvironment struct {
-	ca                      []byte
-	consulClient            *api.Client
-	token                   string
-	policy                  *api.ACLPolicy
-	httpPort                int
-	httpFlattenedPort       int
-	httpReferencePolicyPort int
-	tcpReferencePolicyPort  int
-	grpcPort                int
-	extraHTTPPort           int
-	extraTCPPort            int
-	extraTCPTLSPort         int
-	extraTCPTLSPortTwo      int
-	namespace               string
-	ip                      string
+	ca                               []byte
+	consulClient                     *api.Client
+	token                            string
+	policy                           *api.ACLPolicy
+	httpPort                         int
+	httpFlattenedPort                int
+	httpReferencePolicyPort          int
+	tcpReferencePolicyPort           int
+	parentRefChangeFirstGatewayPort  int
+	parentRefChangeSecondGatewayPort int
+	grpcPort                         int
+	extraHTTPPort                    int
+	extraTCPPort                     int
+	extraTCPTLSPort                  int
+	extraTCPTLSPortTwo               int
```

Comment:
~~Would it be possible to just accept a list of `portsToOpen []int`?~~

Looks like we need a specific reference to each port type for testing, not just to open ports on Kind

Reply:
Yea, all this boilerplate is a significant pain point in writing new e2e tests 😞